canonical/ubuntu-advantage-client
# Handle old config items

This does a couple things:
1. prevents `parse_config` from being called more than once during a simple cli call (e.g. `ua status` unattached)
2. logs a different warning for old config items vs invalid config items
3. only logs warnings for old/invalid config items to the log file (not stdout/stderr)
4. to accomplish (3) it does a minimal version of setup_logging in the middle of `parse_config`

So a few questions to discuss (on Monday at this point most likely):
* Is (2) reasonable long term? Or should we just treat old config items the same as invalid ones?
* Is (3) okay? or do we want to warn louder?
* Is there a better way to do (4)?
* Should we also try to fix the config for the user in postinst or otherwise?

## Test Steps

* in a jammy container
* edit uaclient.conf to include an unknown key (e.g. `bad: bad`) and the old item (i.e. `license_check_log_file: /var/log/log.log`)
* run `ua status`

Without this PR, you'll see a couple warning messages.

With this PR, you won't see any message on the console. But you will see two messages at the top of /var/log/ubuntu-advantage.log

## Desired commit type


## Checklist
